cmake/libprotobuf.cmake

```diff
@@ -46,6 +43,10 @@ set_target_properties(libprotobuf PROPERTIES
     CXX_VISIBILITY_PRESET hidden
     VISIBILITY_INLINES_HIDDEN ON
 )
+get_target_property(OLD_CXX_STANDARD libprotobuf CXX_STANDARD)
+if(OLD_CXX_STANDARD LESS 17)
+  set_target_properties(libprotobuf PROPERTIES CXX_STANDARD 17)
```

Comment:
I think this might be redundant.  We already set `target_compile_features(* PUBLIC cxx_std_17)` [here](https://github.com/protocolbuffers/protobuf/blob/47183273b6fc5ce931888f852dd2a02a7dfaeaef/cmake/protobuf-configure-target.cmake#L10)

If it's not redundant, you could reduce duplication by putting it there.  But I'm not sure what the difference between these things is.  And how would someone else set the CXX_STANDARD property on *our* targets?

Reply:
https://cliutils.gitlab.io/modern-cmake/chapters/features/cpp11.html seems to explain the difference between these 3 options pretty well.  I think our current setup is what we want, except maybe we'd want to mark it `PRIVATE`

Reply:
They shouldn't be PRIVATE because C++17 syntax is leaked in headers.

Reply:
https://cmake.org/cmake/help/v3.20/prop_gbl/CMAKE_CXX_KNOWN_FEATURES.html#high-level-meta-features-indicating-c-standard-support - It appears these are actually *minimums*, not strict settings.  So I think it does make more sense to keep these PUBLIC to make it transitive (see my last comment I resolved)